bcipy/gui/params_form.py

```diff
@@ -1,392 +1,510 @@
 
+from bcipy.helpers.parameters import Parameters
 
-def font(size: int = 14, font_family: wx.Font = wx.FONTFAMILY_SWISS) -> wx.Font:
+Parameter = namedtuple('Parameter', [
+    'value', 'section', 'readableName', 'helpTip', 'recommended_values', 'type'
+])
+
+#--- Utility functions
+
+
+def font(size: int = 14, font_family: str = 'Helvetica') -> QFont:
     """Create a Font object with the given parameters."""
-    return wx.Font(size, font_family, wx.FONTSTYLE_NORMAL, wx.FONTWEIGHT_LIGHT)
+    return QFont(font_family, size, QFont.Normal)
 
 
 def static_text_control(parent,
                         label: str,
                         color: str = 'black',
                         size: int = 14,
-                        font_family: wx.Font = wx.FONTFAMILY_SWISS) -> wx.StaticText:
+                        font_family: str = 'Helvetica') -> QLabel:
     """Creates a static text control with the given font parameters. Useful for
     creating labels and help components."""
-
-    static_text = wx.StaticText(parent, label=label)
-    static_text.SetForegroundColour(color)
-    static_text.SetFont(font(size, font_family))
+    static_text = QLabel(parent)
+    static_text.setWordWrap(True)
+    static_text.setText(label)
+    static_text.setStyleSheet(f'color: {color};')
+    static_text.setFont(font(size, font_family))
     return static_text
 
 
-# Holds the label, help, and input controls for a given parameter.
-FormInput = namedtuple('FormInput', ['control', 'label', 'help'])
-Parameter = namedtuple('Parameter', ['value', 'section', 'readableName',
-                                     'helpTip', 'recommended_values', 'type'])
+class ComboBox(QComboBox):
+    """ComboBox with the same interface as a QLineEdit for getting and setting values.
+
+    Parameters:
+    -----------
+      options - list of items to appear in the lookup
+      selected_value - selected item; if this is not one of the options and new option will be
+        created for this value.
+    """
+
+    def __init__(self, options: List[str], selected_value: str, **kwargs):
+        super(ComboBox, self).__init__(**kwargs)
+        self.options = options
+        self.addItems(self.options)
+        self.setText(selected_value)
+        self.setEditable(True)
+
+    def setText(self, value: str):
+        """Sets the current index to the given value. If the value is not in the list of
+        options it will be added."""
+        if value not in self.options:
+            self.options = [value] + self.options
+            self.clear()
+            self.addItems(self.options)
+        self.setCurrentIndex(self.options.index(value))
+
+    def text(self):
+        """Gets the currentText."""
+        return self.currentText()
 
 
-class Form(wx.Panel):
-    """The Form class is a wx.Panel that creates controls/inputs for each
-    parameter in the provided json file.
+#--- Input widgets
+
+
+class FormInput(QWidget):
```

Comment:
These are generic enough to be used elsewhere and would be useful to the experiment /field UI needs. What do you think about adding the classes to gui_main?

Reply:
I can see how these would be useful in the context of the experiment UI. We probably also want to pull out the method that selects the FormInput (subclass) based on the Parameter type. That code is currently in the ParamsForm widget but could be extracted out into a module function.

Reply:
I agree! If it's not too much effort in this PR let's do it. I'm okay making another ticket otherwise.